cmd/charmd/main.go

```diff
@@ -94,7 +97,37 @@ func serve(confPath string) error {
 	}
 
 	ring := bakery.NewPublicKeyRing()
-	ring.AddPublicKeyForLocation(cfg.IdentityLocation, false, conf.IdentityPublicKey)
+	if conf.IdentityPublicKey != nil {
```

Comment:
How about factoring out the duplicated code below into a separate function, say:

    func addPublicKey(ring bakery.PublicKeyRing, loc string, key *bakery.PublicKey) error

?

or alternatively, you could use httpbakery.PublicKeyRing.

e.g. 
    cache := bakery.NewPublicKeyRing()
    if cfg.IdentityPublicKey != nil {
        cache.AddPublicKeyForLocation(conf.IdentityLocation, false, conf.IdentityPublicKey)
   }
   cfg.PublicKeyLocator = httpbakery.NewPublicKeyRing(http.DefaultClient, cache)